src/metadataTextTracks.js

```diff
@@ -0,0 +1,128 @@
+/**
+* This feature allows metadata text tracks to be manipulated once available, @see process.
+* It also allows ad implementations to leverage ad cues coming through
+* text tracks, @see processAdTrack
+**/
+
+import videojs from 'video.js';
+
+const metadataTextTracks = {};
+
+/**
+* This feature allows metadata text tracks to be manipulated once they are available,
+* usually after the 'loadstart' event is observed on the player
+* @param player A reference to a player
+* @param processTrack A method that performs some operations on a metadata text track
+**/
+metadataTextTracks.process = function(player, processTrack) {
+  const tracks = player.textTracks();
+  const prepareTrack = function(track) {
+    if (track.kind === 'metadata') {
+      // Make sure track is 'hidden' rather than 'disabled' as it may have already
+      // existed and had been 'disabled' during a playlist change
+      track.mode = 'hidden';
+      processTrack(player, track);
+    }
+  };
+
+  // Text tracks are available
+  if (tracks.length > 0) {
+    for (let i = 0; i < tracks.length; i++) {
+      const track = tracks[i];
+      prepareTrack(track);
```

Comment:
Not necessarily. However, some tracks could be disabled but shouldn't be re-enabled while some might be ok to re-enable. This is particularly an issue because if you add a text track via `addTextTrack` you are unable to remove it, so, the only thing you can do (beyond removing the video element and creating a new one) is to just disable the track.

May not necessarily be something that we handle in code but maybe only in documentation. But it's important to know so we can make an informed decision.